### DISCUSSION TITLE
Disabling compression in test mode by default in Rails

### DISCUSSION THREAD

**Discussion by janko** [ElMassimo/vite_ruby#433] [Ideas]
When migrating from Webpacker to Vite, I was comparing compilation speeds in system tests, and Webpacker took ~5s to rebuild asset changes, while Vite took ~8s. I then profiled Vite build, and saw that almost half of the time goes to compression.

After a lot of time trying to figure out where the compression is enabled, I finally found that it's `vite-plugin-rails` that enables it. I then disabled it for test mode:

```js
import { defineConfig } from "vite"
import RailsPlugin from "vite-plugin-rails"

export default defineConfig(({ mode }) => {
  return {
    plugins: [
      RailsPlugin({
        compress: mode !== "test"
      }),
    ]
  }
})
```

This sped up asset rebuilding from ~8s to ~4s. I think that's a significant improvement for the test feedback loop, so I wanted to suggest that `vite-plugin-rails` could disable it in tests by default.

I'm absolutely loving the Vite Ruby ecosystem BTW, I really appreciate all the work you put into it, it's been a pretty smooth journey so far ❤️ I feel like I'm removing a lot of complexity with Vite, my `yarn.lock` is 3x smaller compared to Webpacker setup 😛

**Comment by ElMassimo** [comment]
Hi Janko!

Thanks for the detailed explanation. I think disabling compression in tests is a better default.

Pull requests are welcome! Otherwise I'll implement this next time I'm working in the library.
